lib/game_logic.rb

```diff
@@ -0,0 +1,209 @@
+require_relative '../bin/main'
+
+include UserInteraction
+#====================================================================
+# This is the class Player and is called every time a new game starts
+#====================================================================
+class Player
+    public
+    def validate_rol
+        get_rol
+        if @rol === 1 || @rol === 2
+            board = Board.new ["1","2","3","4","5","6","7","8","9"], @rol, @name_1, @name_2
+            board.print_board
+            
+        else
+            puts "\n\n\nNumber #{@rol} is not a valid option"
+            validate_rol
+        end
+    end
+end
+
+
+
+#===============================================================================================================================================
+# This is the class Rules. This class contains all rules of the game, so everytime a player plays, this class will verify if movements are valid
+# This class also checks if there is a winner or in case the game ends it will puts Game Over
+#===============================================================================================================================================
+
+class Rules < Player
+    def winner?
+        #Horizontal Test
+        
+        if @board[0] === @draw && @board[1] === @draw && @board[2] === @draw
```

Comment:
where does `@board` come from? Same thing for `@draw`?
Where did you declare them? Initialize them?